### PR TITLE
feat: Optimization for setupFormInit

### PR DESCRIPTION
**Pull request by suyashpatil78**
method added to category service for fetching id specific category

Clickup Link - https://app.clickup.com/t/85zrtz58p

### CODE REVIEW

File: src/app/core/services/categories.service.ts [hunk 1]
```diff
@@ -167,4 +167,12 @@ export class CategoriesService {
     const flightSystemCategories = ['Airlines'];
     return flightSystemCategories;
   }
+
+  getCategoryById(id: number): Observable<OrgCategory> {
```
Comment (yksolanki9) [inline src/app/core/services/categories.service.ts:171]: Add unit tests to this method. Check how tests are written for other methods, I can help if needed

**Review by yksolanki9 — CHANGES_REQUESTED** [review]
Add unit test for the new method and resolve Dhar's comments, rest lgtm

File: src/app/core/mock-data/platform-api-category.data.ts [hunk 2]
```diff
@@ -79,3 +79,23 @@ export const platformApiAllCategories: ApiV2Response<PlatformCategory> = {
   ],
   offset: 0,
 };
+
+export const platformApiCategoryById: ApiV2Response<PlatformCategory> = {
+  count: 1,
+  data: [
+    {
```
Comment (yksolanki9) [inline src/app/core/mock-data/platform-api-category.data.ts:86]: Add 2 more categories with different id

File: src/app/core/services/categories.service.spec.ts [hunk 3]
```diff
@@ -145,4 +151,19 @@ describe('CategoriesService', () => {
...
+
+  it('getCategoryById(): should get a category from the api based on ID', (done) => {
+    spenderPlatformV1ApiService.get.and.returnValue(of(platformApiCategoryById));
+    spyOn(categoriesService, 'transformFrom').and.returnValue(transformedOrgCategoryById);
+    spyOn(categoriesService, 'addDisplayName').and.returnValue(expectedOrgCategoryById);
+
+    const categoryId = 141295;
+
+    categoriesService.getCategoryById(categoryId).subscribe((res) => {
+      expect(res).toEqual(expectedOrgCategoryById[0]);
```
Comment (yksolanki9) [inline src/app/core/services/categories.service.spec.ts:163]: use object instead of accessing first value of array

**Review by yksolanki9 — CHANGES_REQUESTED** [review]
Add ss for code coverage to confirm if all branches are covered

**Comment by suyashpatil78** [comment]
Code Coverage - 
image

**Review by yksolanki9 — APPROVED** [review]

**Review by Julias0 — APPROVED** [review]

**Pull request merged**